Source/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -93,6 +95,38 @@ public ISetupSequentialResult<TResult> Returns(TResult value)
 			this.setup.AddReturns(value);
 			return this;
 		}
+		public ISetupSequentialResult<TResult> Returns(Delegate valueFunction)
+		{
+			// If `TResult` is `Delegate`, that is someone is setting up the return value of a method
+			// that returns a `Delegate`, then we have arrived here because C# picked the wrong overload:
+			// We don't want to invoke the passed delegate to get a return value; the passed delegate
+			// already is the return value.
+			if (typeof(TResult) == typeof(Delegate))
+			{
+				return this.Returns(() => (TResult)(object)valueFunction);
+			}
+
+			// The following may seem overly cautious, but we don't throw an `ArgumentNullException`
+			// here because Moq has been very forgiving with incorrect `Returns` in the past.
+			if (valueFunction == null)
+			{
+				return this.Returns(() => default(TResult));
+			}
+
+			if (valueFunction.GetMethodInfo().ReturnType == typeof(void))
+			{
+				throw new ArgumentException(Resources.InvalidReturnsCallbackNotADelegateWithReturnType, nameof(valueFunction));
+			}
+
+			this.setup.AddReturns(valueFunction);
+			return this;
+		}
+
+		public ISetupSequentialResult<TResult> Returns(Func<TResult> valueExpression)
+		{
```

Comment:
Like I mentioned in the issue, I'm not sure if it's 100 % safe to add a new overload to that method. What if there's user code out there that passes a `Func` to the previous only method with the intent that the `Func` itself is the result? If we now add a method overload that takes a `Func`, the compiler might suddenly pick that new overload instead of the original function, causing the sequence setup to produce a different value (the result of the function instead of the function itself). Perhaps try to write a test that provokes this situation.